common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -276,16 +322,47 @@ public String decrypt(final String encryptedBlob)
         // that IV.
         // It is using same cipher for different version since version# change
         // will mean upgrade to AndroidKeyStore and new Key.
-        cipher.init(Cipher.DECRYPT_MODE, secretKey, new IvParameterSpec(bytes, ivIndex,
-                DATA_KEY_LENGTH));
+        cipher.init(
+                Cipher.DECRYPT_MODE,
+                secretKey,
+                new IvParameterSpec(bytes, ivIndex, DATA_KEY_LENGTH)
+        );
 
         // Decrypt data bytes from 0 to ivindex
-        final String decrypted = new String(cipher.doFinal(bytes, KEY_VERSION_BLOB_LENGTH,
-                encryptedLength), AuthenticationConstants.ENCODING_UTF8);
+        final String decrypted = new String(
+                cipher.doFinal(
+                        bytes,
+                        KEY_VERSION_BLOB_LENGTH,
+                        encryptedLength
+                ),
+                AuthenticationConstants.ENCODING_UTF8
+        );
+
         Log.v(TAG, "Finished decryption");
+
         return decrypted;
     }
 
+    private void validateEncodeVersion(String encryptedBlob, int encodeVersionLength) {
+        if (encodeVersionLength <= 0) {
+            throw new IllegalArgumentException(
+                    String.format(
+                            "Encode version length: '%s' is not valid, it must be greater of equal to 0",
+                            encodeVersionLength
+                    )
+            );
+        }
+
+        if (!encryptedBlob.substring(1, 1 + encodeVersionLength).equals(ENCODE_VERSION)) {
+            throw new IllegalArgumentException(
+                    String.format(
+                            "Encode version received was: '%s', Encode version supported is: '%s'",
+                            encryptedBlob,
```

Comment:
>encryptedBlob [](start = 28, length = 13)

Should we be dumping this in the exception message?

Reply:
We should not, will remove. Thanks

Reply:
Resolved in 05bfd4cc9398988713ed8f8b55e232aa6dc83ad9